services/topology-engine-rest/app/app/topology.py

```diff
@@ -613,6 +613,28 @@ def api_v1_topology_link_bandwidth(src_switch, src_port):
     return neo4j_connect.data(query)[0]['r.available_bandwidth']
 
 
+@application.route('/api/v1/topology/routes/src/<src_switch>/dst/<dst_switch>')
+@login_required
+def api_v1_routes_between_nodes(src_switch, dst_switch):
+    depth = request.args.get('depth') or ''
+    query = (
+        "MATCH p="
+        "(src:switch{{name:'{}'}})-[:isl*..{}]->(dst:switch{{name:'{}'}}) "
+        "WITH NODES(p) AS switches "
+        "WHERE ALL(sw1 IN switches "
+        "   WHERE SIZE(FILTER(sw2 IN switches WHERE sw1.name=sw2.name)) = 1) "
+        "RETURN switches"
+    ).format(src_switch, depth, dst_switch)
+
+    try:
+        result = neo4j_connect.data(query)
+
+        paths = [map(format_switch, x['switches']) for x in result]
+        return jsonify(paths)
+    except Exception as e:
+        return "error: {}".format(str(e))
```

Comment:
This "line" lead to response with status code 200 and broken json in body. You must return/raise someone who inherit HTTPException... InternalServerError looks most suitable here.